osu.Game/Overlays/Settings/Sections/Graphics/RendererSettings.cs

```diff
@@ -1,28 +1,39 @@
 ﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
-#nullable disable
-
+using System.Linq;
 using osu.Framework.Allocation;
 using osu.Framework.Configuration;
 using osu.Framework.Graphics;
 using osu.Framework.Localisation;
 using osu.Framework.Platform;
 using osu.Game.Configuration;
 using osu.Game.Localisation;
+using osu.Game.Overlays.Dialog;
 
 namespace osu.Game.Overlays.Settings.Sections.Graphics
 {
     public partial class RendererSettings : SettingsSubsection
     {
         protected override LocalisableString Header => GraphicsSettingsStrings.RendererHeader;
 
+        private bool automaticRendererInUse;
+
         [BackgroundDependencyLoader]
-        private void load(FrameworkConfigManager config, OsuConfigManager osuConfig)
+        private void load(FrameworkConfigManager config, OsuConfigManager osuConfig, IDialogOverlay dialogOverlay, OsuGame game, GameHost host)
         {
-            // NOTE: Compatability mode omitted
+            var renderer = config.GetBindable<RendererType>(FrameworkSetting.Renderer);
+            automaticRendererInUse = renderer.Value == RendererType.Automatic;
+
             Children = new Drawable[]
             {
+                new SettingsEnumDropdown<RendererType>
+                {
+                    LabelText = GraphicsSettingsStrings.Renderer,
+                    Current = renderer,
+                    Items = host.GetPreferredRenderersForCurrentPlatform().OrderBy(t => t),
+                    Keywords = new[] { @"compatibility", @"directx" },
+                },
```

Comment:
The dropdown needs to be hidden on android for now. Otherwise users can get themselves in a broken state which is unrecoverable without manually editiing `framework.ini`.